### PR TITLE
wip Add upload and create method for boards

### PR DESCRIPTION
**Pull request by davidenko87**
## Proposed changes

As requested by @arnoFleming(#36) , the simplest way to create a board with one method. 
Also added some shortcuts for getting file or links only. 

## Types of changes

- [x] Docs (missing or updated docs)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement / maintenance (removing technical debt, performance, tooling, etc...)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/WeTransfer/wetransfer_ruby_sdk/blob/master/.github/CONTRIBUTING.md) doc
- [x] I have added necessary documentation (if appropriate)
- [x] Code changes are covered by unit tests.
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: spec/we_transfer_client/remote_board_spec.rb [hunk 1]
```diff
@@ -68,6 +68,20 @@
     end
   end
 
+  describe '#link_items' do
+    it 'returns only file items' do
```
Comment (arnoFleming) [inline spec/we_transfer_client/remote_board_spec.rb]: `s/file/link/`

File: spec/we_transfer_client/remote_board_spec.rb [hunk 2]
```diff
@@ -68,6 +68,20 @@
...
+      expect(subject.items.size).to eq(2)
+      expect(subject.file_items.size).to eq(1)
```
Comment (arnoFleming) [inline spec/we_transfer_client/remote_board_spec.rb]: False positive: `s/file/link`

File: lib/we_transfer_client/transfers.rb [hunk 3]
```diff
@@ -2,11 +2,24 @@ module WeTransfer
   class Client
     module Transfers
 
+      def create_transfer_and_upload_files(message:, transfer_builder_class: TransferBuilder, future_transfer_class: FutureTransfer)
+        builder = transfer_builder_class.new
+        yield(builder)
+        future_transfer = future_transfer_class.new(message: message, files: builder.items)
+        transfer = create_remote_transfer(future_transfer)
+        transfer.files.each do |item|
```
Comment (arnoFleming) [inline lib/we_transfer_client/transfers.rb]: please rename your block param from `item` to `file`

File: lib/we_transfer_client/transfers.rb [hunk 4]
```diff
@@ -2,11 +2,24 @@ module WeTransfer
...
+          local_file = future_transfer.files.select { |x| x.name == item.name }.first
```
Comment (arnoFleming) [inline lib/we_transfer_client/transfers.rb]: Don't know about the domain, but could we upload 2 files with the same name, and do we ~want to~ need to do something here?

File: lib/we_transfer_client/future_board.rb [hunk 5]
```diff
@@ -7,6 +7,14 @@ def initialize(name:, description: nil, items: [])
     @items = items
   end
 
+  def file_items
```
Comment (arnoFleming) [inline lib/we_transfer_client/future_board.rb]: Could you able to find an abstraction for 4 methods (in remote and future), so it involves less duplication?

File: lib/we_transfer_client.rb [hunk 6]
```diff
@@ -55,6 +55,7 @@ def put_io_in_parts(object:, file:, io:)
         )
         ensure_ok_status!(response)
       end
+      {success: true, message: 'File Uploaded'}
```
Comment (arnoFleming) [inline lib/we_transfer_client.rb]: This response makes sense to me, but I don't see it being used. Or do I overlook something?

File: lib/we_transfer_client/boards.rb [hunk 7]
```diff
@@ -2,6 +2,19 @@ module WeTransfer
   class Client
     module Boards
 
+      def create_board_and_upload_files(name:, description:, board_builder: BoardBuilder, future_board_class: FutureBoard)
```
Comment (arnoFleming) [inline lib/we_transfer_client/boards.rb]: Please use the methods that the class already provides

Comment (arnoFleming) [reply lib/we_transfer_client/boards.rb]: @davidenko87 and I discussed this, it is being taken care of.

File: lib/we_transfer_client.rb (hunk 6 above)
Comment (davidenko87) [reply lib/we_transfer_client.rb]: it's a response for the user, haven't figured out what the actual response should be. Could also be a simple true

**Pull request merged**